pkg/cvo/upgradeable.go

```diff
@@ -442,3 +449,31 @@ func (optr *Operator) adminGatesEventHandler() cache.ResourceEventHandler {
 		DeleteFunc: optr.deleteFunc,
 	}
 }
+
+// shouldSyncUpgradeableDueToPreconditionChecks checks if the upgradeable status should
+// be synchronized due to the precondition checks. It checks whether the precondition
+// checks on the payload are failing for less than minimumUpdateCheckInterval, and it has
+// been more than the statusInterval since the last synchronization. This means we will
+// synchronize the upgradeable status more frequently at beginning of an upgrade for
+// a while in a case the precondition checks are failing.
+//
+// shouldSyncUpgradeableDueToPreconditionChecks expects the parameters not to be nil.
+//
+// Function returns true if the synchronization should happen, returns false otherwise.
+//
+// A precondition check preventing an upgrade could have been potentially already resolved
+// (e.g. operator reporting Upgradeable==false just momentarily).
+// CVO should synchronize the upgradeable status again in case of failing precondition
+// checks to speed up initializing an upgrade stuck on a precondition check that
+// potentially has been solved.
+//
+// We don't want to check this forever in case of precondition checks failing for a long
+// time due to a bigger problem that won't resolve by itself in a timely manner.
```

Comment:
This was already explained so I would drop.

Reply:
Dropped in the new commit https://github.com/openshift/cluster-version-operator/commit/5aac7a3ef2934cbeae09a2632e22fe57eb20da91.